explore/realspace.py

```diff
@@ -91,11 +93,14 @@ def invert_view(qx, qy, view):
     """
     theta, phi, psi = view
     q = np.vstack((qx.flatten(), qy.flatten(), 0*qx.flatten()))
-    return np.asarray((Rz(-psi)*Ry(-theta)*Rz(-phi))*np.matrix(q))
+    # CRUFT: py3 allows ((Rz(phi) @ Ry(theta) @ Rz(psi)) @ q
+    Rinv = dot(dot(Rz(-psi), Ry(-theta)), Rz(-phi))
```

Comment:
`a*b*c` expands to `mul(mul(a, b), c)` which for `np.matrix` is equivalent to `dot(dot(a, b), c)`. With the addition of the `@` operator numpy is deprecating the `matrix` class, hence the change. The change itself doesn't warrant a unit test. Whether there should be unit tests on this validation code is a separate question.

I'm using `dot` rather than `@` since sasmodels is still supporting python 2.7 for no particularly good reason.

Reply:
Core mathematical functions are where you need to have the highest trust, so that's where you want your tests - especially when it's unavoidably hard to read.